File: 0746 - Min Cost Climbing Stairs/solution.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& A) {
        int n = A.size();
        for(int i = 2; i < n; i++)
            A[i] += min(A[i-1], A[i-2]);
        return min(A[n-1], A[n-2]);
    }
};

File: 0746 - Min Cost Climbing Stairs/solution.py
class Solution:
    def minCostClimbingStairs(self, A: List[int]) -> int:
        @lru_cache(None)
        def dp(n):
            if n < 0:
                return 0
            return A[n]+min(dp(n-1), dp(n-2))
        return min(dp(len(A)-1), dp(len(A)-2))